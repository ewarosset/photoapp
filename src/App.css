.websiteHeader{
    display: flex;
    justify-content: center;
}