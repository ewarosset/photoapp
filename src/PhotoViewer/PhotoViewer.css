
.mainPhoto{
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 50%;
    border: 3px solid transparent;
    border-radius: 2%;
}

.thumbnails img
{
    width: 25%;
    padding: 1px;
    border: 3px solid transparent;
    border-radius: 3%;
}

.thumbnails img:hover
{
    border-color:yellow;
}

.thumbnails{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    width: 50%;
    margin: auto;
}

.selectImage{
    margin-right: auto;
    margin-left: auto;
    width: 50%;
    border: 3px solid transparent;
}

