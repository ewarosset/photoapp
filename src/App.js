import logo from './logo.svg';
import doggo from './dog-food.jpg'
import './App.css';
import {PhotoViewer} from "./PhotoViewer/PhotoViewer";
import {Clicker} from "./Clicker/Clicker";
import {imageUrls} from "./PhotoViewer/getImages"
import React, {useEffect} from "react";
import { hotjar } from 'react-hotjar';
import {BrowserRouter as Router, Route, Routes} from 'react-router-dom';
import Homepage from "./Pages/Homepage/Homepage";
import TestPage from "./Pages/Page2/TestPage";
import ReactGA from 'react-ga';

// const hotjarTest = () => {
//     hotjar.identify('USER_ID', { userProperty: 'value' });
//     hotjar.event('test-button');
//     hotjar.stateChange('/my/page');
// };

function App() {

    useEffect(() => {
        hotjar.initialize(2813391, 6);
    }, []);

    useEffect(() => {
        ReactGA.initialize('UA-219300525-2');
    }, []);

    return (
        <Router>
            <Routes>
                <Route path="photoapp" element={<Homepage/>}/>
                <Route path="otherpage" element={<TestPage/>}/>
            </Routes>
        </Router>
    );
}

export default App;
