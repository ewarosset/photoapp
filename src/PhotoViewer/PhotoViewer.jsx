import React, {useState} from "react";  // import React (to provide access to JSX)
import './PhotoViewer.css';
import {imageUrls} from './getImages'


export function PhotoViewer() {

    const [currentImage, setImage] = useState("https://picsum.photos/id/600/1600/900.jpg")

    

    return (
        <div>
            <img className="mainPhoto" alt='mainPic' src={currentImage}/>
            <h2 className={"selectImage"}>Select your photo</h2>
            {/*<DisplayThumbnails setImageFunction={setImage} selectedImage={currentImage}/>*/}
            
            
            <div className="thumbnails">
                {imageUrls.map((image, index) => (
                    
                    <img src={image} alt={image} onClick={() => setImage(image)}/>
                ))
                }
            </div>
        </div>
    );
}

// function DisplayThumbnails(props) {
//
//     let allImages = [];
//     for (let url of imageUrls)
//     {
//         allImages.push(
//             <img className="thumbnailImages"  onClick={() => props.setImageFunction(url)} src={url} alt={url}/>
//         );
//     }
//    
//     return (
//         <div className="thumbnails">{allImages}</div>
//     )
// }

