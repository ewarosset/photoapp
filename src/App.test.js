import {fireEvent, getByAltText, render, screen} from '@testing-library/react';
import App from './App';
import {imageUrls} from "./PhotoViewer/getImages"
import {PhotoViewer} from "./PhotoViewer/PhotoViewer";
import renderer from 'react-test-renderer';


test('check heading is included', () =>{
  render(<App />);
  const heading = screen.getByText(/React Photo Viewer/)
  expect(heading).toBeInTheDocument();
})


test('test image url generation are not null', () => {
  expect(imageUrls).not.toBeNull();
});

test('test image url generation contains random image', () => {
  expect(imageUrls).toContain("https://picsum.photos/id/603/1600/900.jpg");
});


test('set Image function changes the image', () =>{

  render(<PhotoViewer />);

  fireEvent.click(screen.getByAltText('https://picsum.photos/id/602/1600/900.jpg'));

  const image = screen.getByAltText('mainPic');

  expect(image).toHaveAttribute('src', 'https://picsum.photos/id/602/1600/900.jpg')

})


test('Ensure PhotoViewer is the same', () => {
  const tree = renderer
      .create(<PhotoViewer />)
      .toJSON();
  expect(tree).toMatchSnapshot();
});

