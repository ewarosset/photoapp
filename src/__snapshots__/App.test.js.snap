// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure PhotoViewer is the same 1`] = `
<div>
  <img
    alt="mainPic"
    className="mainPhoto"
    src="https://picsum.photos/id/600/1600/900.jpg"
  />
  <h2
    className="selectImage"
  >
    Select your photo
  </h2>
  <div
    className="thumbnails"
  >
    <img
      alt="0https://picsum.photos/id/600/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/600/1600/900.jpg"
    />
    <img
      alt="1https://picsum.photos/id/602/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/602/1600/900.jpg"
    />
    <img
      alt="2https://picsum.photos/id/603/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/603/1600/900.jpg"
    />
    <img
      alt="3https://picsum.photos/id/604/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/604/1600/900.jpg"
    />
    <img
      alt="4https://picsum.photos/id/605/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/605/1600/900.jpg"
    />
    <img
      alt="5https://picsum.photos/id/606/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/606/1600/900.jpg"
    />
    <img
      alt="6https://picsum.photos/id/607/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/607/1600/900.jpg"
    />
    <img
      alt="7https://picsum.photos/id/608/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/608/1600/900.jpg"
    />
    <img
      alt="8https://picsum.photos/id/609/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/609/1600/900.jpg"
    />
    <img
      alt="9https://picsum.photos/id/610/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/610/1600/900.jpg"
    />
    <img
      alt="10https://picsum.photos/id/611/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/611/1600/900.jpg"
    />
    <img
      alt="11https://picsum.photos/id/612/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/612/1600/900.jpg"
    />
    <img
      alt="12https://picsum.photos/id/613/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/613/1600/900.jpg"
    />
    <img
      alt="13https://picsum.photos/id/614/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/614/1600/900.jpg"
    />
    <img
      alt="14https://picsum.photos/id/615/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/615/1600/900.jpg"
    />
    <img
      alt="15https://picsum.photos/id/616/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/616/1600/900.jpg"
    />
    <img
      alt="16https://picsum.photos/id/617/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/617/1600/900.jpg"
    />
    <img
      alt="17https://picsum.photos/id/618/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/618/1600/900.jpg"
    />
    <img
      alt="18https://picsum.photos/id/619/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/619/1600/900.jpg"
    />
    <img
      alt="19https://picsum.photos/id/620/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/620/1600/900.jpg"
    />
    <img
      alt="20https://picsum.photos/id/621/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/621/1600/900.jpg"
    />
    <img
      alt="21https://picsum.photos/id/622/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/622/1600/900.jpg"
    />
    <img
      alt="22https://picsum.photos/id/623/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/623/1600/900.jpg"
    />
    <img
      alt="23https://picsum.photos/id/625/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/625/1600/900.jpg"
    />
    <img
      alt="24https://picsum.photos/id/626/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/626/1600/900.jpg"
    />
    <img
      alt="25https://picsum.photos/id/627/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/627/1600/900.jpg"
    />
    <img
      alt="26https://picsum.photos/id/628/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/628/1600/900.jpg"
    />
    <img
      alt="27https://picsum.photos/id/629/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/629/1600/900.jpg"
    />
    <img
      alt="28https://picsum.photos/id/630/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/630/1600/900.jpg"
    />
    <img
      alt="29https://picsum.photos/id/631/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/631/1600/900.jpg"
    />
    <img
      alt="30https://picsum.photos/id/633/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/633/1600/900.jpg"
    />
    <img
      alt="31https://picsum.photos/id/634/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/634/1600/900.jpg"
    />
    <img
      alt="32https://picsum.photos/id/635/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/635/1600/900.jpg"
    />
    <img
      alt="33https://picsum.photos/id/637/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/637/1600/900.jpg"
    />
    <img
      alt="34https://picsum.photos/id/638/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/638/1600/900.jpg"
    />
    <img
      alt="35https://picsum.photos/id/639/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/639/1600/900.jpg"
    />
    <img
      alt="36https://picsum.photos/id/640/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/640/1600/900.jpg"
    />
    <img
      alt="37https://picsum.photos/id/641/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/641/1600/900.jpg"
    />
    <img
      alt="38https://picsum.photos/id/642/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/642/1600/900.jpg"
    />
    <img
      alt="39https://picsum.photos/id/643/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/643/1600/900.jpg"
    />
    <img
      alt="40https://picsum.photos/id/645/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/645/1600/900.jpg"
    />
    <img
      alt="41https://picsum.photos/id/646/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/646/1600/900.jpg"
    />
    <img
      alt="42https://picsum.photos/id/648/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/648/1600/900.jpg"
    />
    <img
      alt="43https://picsum.photos/id/649/1600/900.jpg"
      onClick={[Function]}
      src="https://picsum.photos/id/649/1600/900.jpg"
    />
  </div>
</div>
`;
